### DISCUSSION TITLE
About Usage Statistics

### DISCUSSION THREAD

**Discussion by jcarnat** [searxng/searxng#3643] [Q&A] [answered]
Hi,

I´d like to get some (anonymous) usage stats of my public instance.
What I'm looking for is a way to get:
- how many incoming search requests were made ;
- how many outgoing queries were made.

The "/stats" URL does not seem to be accurate as I don´t see numbers growing as I make queries. I've read this may be due to the fact that stats are maintained by workers - I have 1 master and 4 worker, using gunicorn.

I've tried looking at system's incoming and outgoing TCP connexions on port 443. But those include incoming health-checks and outgoing maintenance connexions. So not really accurate either.

I've looked at the code and the Redis content but I couldn´t understand if those numbers are available there.

Is there a way to get those statistics out of searxng?

**Comment by return42** [comment, marked as answer]
>  I've read this may be due to the fact that stats are maintained by workers - I have 1 master and 4 worker, using gunicorn.

Yes and when restarting workers every counter is gone.

> Is there a way to get those statistics out of searxng?

Sadly not .. the counters only in-process (workers) and we do not have strategy what actions should how long be counted.

**Reply by jcarnat** [reply]
Ok. I'll try to find another way them. Thank you.

**Comment by vojkovic** [comment]
See https://github.com/searxng/searxng/pull/3157

**Reply by jcarnat** [reply]
Yes, I've seen it while looking for an answer. But AFAIK, this is not enabled (yet?).

**Reply by vojkovic** [reply]
@jcarnat Yes - this feature has not been merged yet.

I'm not entirely sure why considering it's a pretty useful feature for both users and administrators.

I'm unable to see how this feature could realistically have an effect on user privacy - but that's a discussion for the SearXNG team.

**Reply by return42** [reply]
> I'm not entirely sure why considering it's a pretty useful feature for both users and administrators.

Sorry its just lack of time .. BTW its only the first step since:

> the counters only in-process (workers) and we do not have strategy what actions should how long be counted.
